components/encoder/single/single_encoder.go

```diff
@@ -182,10 +184,13 @@ func (e *Encoder) Reconfigure(
 // Start starts the Encoder background thread.
 func (e *Encoder) Start(ctx context.Context) {
 	encoderChannel := make(chan board.Tick)
-	e.I.AddCallback(encoderChannel)
+	err := e.board.StreamTicks(e.cancelCtx, []string{e.diPinName}, encoderChannel, nil)
+	if err != nil {
+		utils.Logger.Errorw("error getting interrupt ticks", "error", err)
+		return
+	}
 	e.activeBackgroundWorkers.Add(1)
 	utils.ManagedGo(func() {
-		defer e.I.RemoveCallback(encoderChannel)
```

Comment:
Same question here about how the board knows to remove the callbacks later. This is important: if I create one of these encoders, realize I've done it wrong, delete the old one and create a new correct one, I don't want the interrupts to lock up trying to communicate with the old one afterwards.